Skip index verification for not completed constraints during the consistency check

MishaDemianenko opened the pull request:
Constraints create using separate transactions and separate commands.
It can be the case that index command was already executed and the index is there
while actual constraint transaction is not yet/longer there.
This PR updates only consistency checker to ignore such indexes and
only warn about those instead of claim about store inconsistency.

MishaDemianenko committed:
Skip index verification for not completed constraints during the consistency check

Constraints create using separate transactions and separate commands.
It can be the case that index command was already executed and the index is there
while actual constraint transaction is not yet/longer there.
This PR updates only consistency checker to ignore such indexes and
only warn about those instead of claim about store inconsistency.

tinwelint commented on community/kernel/src/main/java/org/neo4j/kernel/impl/store/record/IndexRule.java:
```diff
@@ -114,6 +114,11 @@ public Long getOwningConstraint()
         return owningConstraint;
     }
 
+    public boolean isNotValidatedIndexConstraint()
```
As with all boolean methods it's way nicer if they are positive in nature... easier to grok when read in conditions.

tinwelint reviewed (changes requested):
Looks good, just that inverted boolean method, that's all

MishaDemianenko committed:
Method rename
indexRule.isNotValidatedIndexConstraint -> indexRule.isIndexWithoutOwningConstraint

MishaDemianenko commented:
@tinwelint updated

MishaDemianenko commented:
@tinwelint ping ping

tinwelint reviewed (approved):
it still has a negative tone to it, i.e. `without`. It wasn't the name in particular, more that it should be flipped, true/false wise. But I won't bike-shed this, it pretty enough!
